src/worker/smartOrderRouter.ts

```diff
@@ -0,0 +1,235 @@
+import DEFAULT_TOKEN_LIST from '@uniswap/default-token-list'
+import { BigintIsh, Currency, CurrencyAmount, Token, TradeType } from '@uniswap/sdk-core'
+import {
+  AlphaRouter,
+  AlphaRouterConfig,
+  CachingGasStationProvider,
+  CachingTokenListProvider,
+  EIP1559GasPriceProvider,
+  HeuristicGasModelFactory,
+  ICache,
+  ID_TO_CHAIN_ID,
+  IMetric,
+  MetricLoggerUnit,
+  PoolProvider,
+  QuoteProvider,
+  routeAmountsToString,
+  setGlobalMetric, SwapRoute, UniswapMulticallProvider,
+  URISubgraphProvider
+} from '@uniswap/smart-order-router'
+import * as Comlink from 'comlink'
+import { NETWORK_URLS } from 'connectors/constants'
+import { ethers } from 'ethers'
+import JSBI from 'jsbi'
+import { GetQuoteResult } from 'state/routing/types'
+
+type TokenInRoute = Pick<Token, 'address' | 'chainId' | 'symbol' | 'decimals'>
+
+type PoolInRoute = {
+  type: 'v3-pool'
+  address: string
+  tokenIn: TokenInRoute
+  tokenOut: TokenInRoute
+  sqrtRatioX96: string
+  liquidity: string
+  tickCurrent: string
+  fee: string
+  amountIn?: string
+  amountOut?: string
+}
+
+class MetricLogger extends IMetric {
+  putDimensions() {
+    return
+  }
+
+  putMetric(key: string, value: number, unit?: MetricLoggerUnit) {
+    console.info({ key, value, unit }, `[Metric]: ${key}: ${value} | ${unit ? unit : ''}`)
+  }
+}
+setGlobalMetric(new MetricLogger())
+
+const DEFAULT_ROUTING_CONFIG: AlphaRouterConfig = {
+  topN: 2,
+  topNDirectSwaps: 2,
+  topNTokenInOut: 3,
+  topNSecondHop: 0,
+  topNWithEachBaseToken: 3,
+  topNWithBaseToken: 6,
+  topNWithBaseTokenInSet: false,
+  maxSwapsPerPath: 3,
+  minSplits: 1,
+  maxSplits: 7,
+  distributionPercent: 5,
+}
+
+class Cache<T> implements ICache<T> {
+  async get() {
+    return undefined
+  }
+  async set() {
+    return false
+  }
+  async has() {
+    return false
+  }
+}
+
+const chainId = 1
+const provider = new ethers.providers.JsonRpcProvider(NETWORK_URLS[chainId])
+
+const subgraphProvider = new URISubgraphProvider(chainId, 'https://ipfs.io/ipfs/QmfArMYESGVJpPALh4eQXnjF8HProSF1ky3v8RmuYLJZT4')
+
+const multicall2Provider =
+  new UniswapMulticallProvider(chainId, provider, 375_000)
+
+const poolProvider = new PoolProvider(ID_TO_CHAIN_ID(chainId), multicall2Provider)
+
+const tokenListProvider = new CachingTokenListProvider(chainId, DEFAULT_TOKEN_LIST, new Cache())
```

Comment:
We should probably use the token list selected by the user, and maybe re-initialize the web worker if they change it